src/components/Incidents/utils.js

```diff
@@ -11,8 +12,6 @@ export function groupAndDeduplicate(objects, keyType) {
       key = obj.metric.alertname + obj.metric.namespace;
     } else if (keyType === 'group_id') {
       key = obj.metric.group_id;
-    } else {
```

Comment:
If this is removed, we would create a new entry with the undefined key for the groupedObjects Map

Reply:
This is not an issue, there are 2 places where this function is called. processAlertTimestamps and processIncidentTimestamps and in both cases we directly pass the string parameter.

Reply:
In that case we can define a type for the `keyType: 'alertname+namespace' | 'group_id'` to make this function more robust at least in compilation time as TypeScript will check that the correct values are passed as params